src/common/prime.c

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2003-2018, Great Software Laboratory Pvt. Ltd.
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <math.h>
+#include "prime.h"
+
+/*****************************************
+ * Return whether value is prime or not
+ *
+ * Returns:
+ *   1  = prime
+ *   0  = not prime
+ *   -1 = undefined (i.e. x < 2)
+*****************************************/
+int onf_is_prime(const int x) {
```

Comment:
This can be a bit faster, see https://en.wikipedia.org/wiki/Primality_test#Pseudocode